File: mock-facility-management/src/main/java/de/vkb/dojo/es/facilityManagement/kafka/config/StoreNames.java
package de.vkb.dojo.es.facilityManagement.kafka.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;

@Configuration
public class StoreNames {
    @Value("${spring.kafka.streams.roomEventAggregator.stores.aggregate}")
    public String roomEventAggregate;
    @Value("${spring.kafka.streams.roomReader.stores.state}")
    public String roomReaderState;
    @Value("${spring.kafka.streams.feedbackReader.stores.state}")
    public String feedbackReaderState;
}

File: mock-human-resources/src/main/kotlin/de/vkb/dojo/es/humanResources/kafka/config/StoreNames.kt
package de.vkb.dojo.es.humanResources.kafka.config

import io.micronaut.context.annotation.ConfigurationProperties
import io.micronaut.context.annotation.Property
import io.micronaut.context.annotation.Value
import jakarta.inject.Singleton

@Singleton
data class StoreNames(
    @Property(name = "kafka.streams.default.stores.aggregate") val personEventAggregate: String,
    @Property(name = "kafka.streams.person-reader.stores.state") val personReaderState: String,
    @Property(name = "kafka.streams.feedback-reader.stores.state") val feedbackReaderState: String
)
